### PR TITLE
Tracks for JITM dismiss success

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #7630 

### Description

This PR has analytics to track when the JITM dismiss success as well as JITM dismiss failure

`*_jitm_dismiss_success`

Properties: 
source : `my_store`
jitm_id : {jitm_id}
jitm_group : {FEATURE_CLASS}

`*_jitm_dismiss_failure`

Properties: 
source : `my_store`
jitm_id : {jitm_id}
jitm_group : {FEATURE_CLASS}
error_code
error_description

More info: pdfdoF-1zc-p2

### Pre-requisite
As mentioned in this post (pdfdoF-1vt-p2) make sure you have installed any one of the mentioned barcode plugin installed on your store. For example - you can install [A4 Barcode Generator](https://wordpress.org/plugins/a4-barcode-generator/)

Your store must be set up in the US. JITM is not supported in any other country.

### JITM dismiss success
1. Open the WooCommerce android app and you should see My Store tab by default as the first screen
2. Make sure you see JITM related to the IPP.
3. Tap on the JITM dismiss button (cross icon)
4. Check in the logs that JITM dismiss success event is tracked

### JITM dismiss failure
1. Open the WooCommerce android app and you should see My Store tab by default as the first screen
2. Make sure you see JITM related to the IPP.
3. Turn off the internet on your device
4. Tap on the JITM dismiss button (cross icon)
5. Check in the logs that JITM dismiss failure event is tracked

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt [hunk 1]
```diff
@@ -250,10 +250,51 @@ class MyStoreViewModel @Inject constructor(
...
+                        featureClass,
+                        response.error?.type?.name,
+                        response.error?.message
+                    )
+                }
+            }
         }
     }
 
+    private fun trackJitmDismissFailureEvent(
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt:268]: 💡 Seems like a lot of the tracking code lives in the VM and the VM has hundreds of lines now - we might want to consider extracting it away.

**Review by malinajirka — APPROVED** [review]
Thanks @AnirudhBhat, the code works as expected. However, I noticed that after I dismiss the banner, it re-appears on the next app launch - is that expected? I'd expect it'd to reappear after a couple weeks or so 😕. (cc @joshheald)

**Comment by pachlava** [comment]
Just leaving a note that Instrumented tests fail in this PR (and other 7 related PRs) because of the new JITM-related request, not yet matched by any of WireMock request matcher files.

I'll start with creating this file, and will get back here once it's done. I'd prefer to have it added as a part of these PRs, otherwise we will get a lot of fail alerts in our `#build-...` channel.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt:268]: Agree! I've created an issue [here](https://github.com/woocommerce/woocommerce-android/issues/7668) to work on it in the next release

**Comment by AnirudhBhat** [comment]
>I noticed that after I dismiss the banner, it re-appears on the next app launch - is that expected? I'd expect it'd to reappear after a couple weeks or so 

This is the expected behavior for our test JITM setup. It will reappear after 30 seconds from the last dismissal.

**Comment by pachlava** [comment]
> I'll start with creating this (Wire Mock) file, and will get back here once it's done.

Just to close the loop - the file was added to #7652 with 20bb09e719857a88cfbed20d36fad89eb01a49cc, and Instrumented tests started to pass after that.

**Pull request merged**